### DISCUSSION TITLE
How "impossible to use" is acceptable before releasing a dev tool?

### DISCUSSION THREAD

**Discussion by tradegear** [dotnet/maui#11208] [General]
I have run into eight fatal flaws that I have posted to this git and the VS Feedback site, and received virtually zero meaningful resolution to any.  I worked around or pushed through six of them, but two of them (1.  cannot get VS to produce/publish a release build that Playstore doesn't flag as being built in "debug mode"  2.  Random crash in iOS version of the app, causing reject by Appstore).   This project should've taken less than 50 hrs, and yet I've spent 8 weeks fighting with this.
Microsoft: Please stop adding new features, and make the stuff that already has been released do RELIABLY what was promised when it was released.  Versluis and Montemagno are great at hyping things up in their Youtube videos, with "great" and "super-cool", but then the stuff just crashes and I've not only wasted my time watching them, but also wasted weeks trying to push through issue after issue until I reach the point of having built a non-deployable app.  Now I have very unhappy clients who think I'm pretty stupid, and I kind of agree with them as I don't get paid for the past eight weeks.  I've been a professional developer for 38 years, mostly using Microsoft tools since around 1991, and this is the worst I've experienced.

**Comment by whann0205** [comment]
I have tried to get my app running since visual studio 2019 and 
.net 3.1 and mobile blazor bindings - more than 2 Years! 
There were always things with stops me from deploying. 
In each version since then something stopped working or needs a complete 
rewrite. Especially related to iphone and ios there is always something wrong with the
pathnames. For god´s sake does anybody at Microsoft tests. 
Most of them i can be reproduced by just compiling a predefined app from their samples site!
But always they force you to make repro project and analyse everything yourself
which costs a lot of time. 
I have not found a way to contact the developers or support outside this github either.
And one might think ok then dont use the newest version for some fancy
things. But the problem is that always something is in PREVIEW and 
cannot be relied and if u use that released version important functions 
(connecting to your database, getting country names, etc.) does not work.
I agree that guys montemagno, eilon and so on are really great developers!
But we need to have a stable thing which less fancy things and more stability!
I begin to wonder about flutter...

**Reply by tradegear** [reply]
Flutter looks pretty interesting.  I see a video snippet with a guy wearing a Google shirt.  Coincidence, or intentional?  Does this mean I get to reduce the number of "players" in this process from 3 (Microsoft, Google, and Apple) to only 2 (Google, and Apple)?  There is a phrase: "only one throat to choke", it is best when there is only one vendor responsible for a solution.  (like IBM in the "old days", when everything was unbreakable and I knew every program problem was MY fault)  But 2 is better than 3.  The font of the Flutter website looks identical to the Google Cloud Platform Console.... maybe not a coincidence.
